src/charm.py

```diff
@@ -419,6 +424,46 @@ def _on_upgrade_charm(self, _event: UpgradeCharmEvent) -> None:
         runner_manager.flush()
         self._reconcile_runners(runner_manager)
 
+    def _configure_aproxy(self, aproxy_proxy: str) -> None:
+        """Configure aproxy.
+
+        Args:
+            aproxy_proxy: Proxy to configure aproxy with.
+        """
+        logger.info("Configuring aproxy for the runner.")
+
+        execute_command(["snap", "set", "aproxy", f"proxy={aproxy_proxy}"])
+
+        stdout, _ = execute_command(
+            [
+                "/bin/bash",
+                "-c",
+                r"ip route get $(ip route show 0.0.0.0/0 | grep -oP 'via \K\S+') |"
+                r" grep -oP 'src \K\S+'",
+            ],
```

Comment:
This script is to get the ip address of the default nic. I think this is obtainable from the dnsmasq DHCP lease record on host `/var/snap/lxd/common/lxd/networks/lxdbr0/dnsmasq.leaseses`

Reply:
Using your method, I get the IP of the runner vm, not the charm unit. But aproxy is installed in the charm unit.

Reply:
Have updated the code

Reply:
Sorry for my lack of understanding, would this and the following code be trying to:
route all non-local http/s traffic from/towards the charm unit to each of the spawned runner VMs to the dnat and hence we need the charm unit IP?

Reply:
So, I moved aproxy inside the runner vm with the latest commits as requested by weii through latest commits https://github.com/canonical/github-runner-operator/pull/147#discussion_r1395426026 .
The goal is to route all non-local http/s traffic from inside the runner to aproxy. We now need the runner IP, not the charm unit IP.